ink-wrapper-types/src/aleph_client.rs

```diff
@@ -1,14 +1,37 @@
-use aleph_client::AsConnection;
-use aleph_client::Balance;
 use aleph_client::{
     pallets::contract::{ContractCallArgs, ContractRpc, ContractsUserApi},
-    SignedConnectionApi, TxInfo, TxStatus,
+    sp_weights::weight_v2::Weight,
+    AsConnection, Balance, CodeHash, ConnectionApi, SignedConnectionApi, TxInfo, TxStatus,
 };
 use anyhow::Error;
 use anyhow::{anyhow, Context, Result};
 use async_trait::async_trait;
 use ink_primitives::AccountId;
 use pallet_contracts_primitives::ContractExecResult;
+use pallet_contracts_primitives::ContractInstantiateResult;
+use scale::Encode;
+use subxt::ext::sp_core::Bytes;
+use subxt::rpc_params;
+
+#[derive(Encode)]
+struct InstantiateRequest {
+    origin: [u8; 32],
+    value: Balance,
+    gas_limit: Option<Weight>,
+    storage_deposit_limit: Option<Balance>,
+    code: Code,
+    data: Vec<u8>,
+    salt: Vec<u8>,
+}
+
+#[derive(Encode)]
+enum Code {
+    /// The Wasm blob to be instantiated.
+    #[allow(dead_code)]
+    Code(Vec<u8>),
```

Comment:
I would focus on creating a tool for generating the type-safe API to _already-deployed_ contracts and leave the contract upload (or more generally - deployment and setup) to an external tool.

Reply:
Yeah, this is only here, to mirror the definition in pallet-contracts-rpc-runtime-api or whatever. I suspect if I don't define it it might get encoded incorrectly with scale.